src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,89 @@
 import { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+type Props = {
+  onAdd: (newMovie: Movie) => void;
+};
+
+const initialMovie = {
+  title: '',
+  description: '',
+  imgUrl: '',
+  imdbUrl: '',
+  imdbId: '',
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const [count, setCount] = useState(0);
+
+  const [movie, setMovie] = useState(initialMovie);
+
+  const isFormFilled
+    = !movie.title.trim()
+    || !movie.imgUrl.trim()
+    || !movie.imdbUrl.trim()
+    || !movie.imdbId.trim();
+
+  const handleInputChange = (key: string, value: string) => {
+    setMovie((prevInputs) => ({ ...prevInputs, [key]: value }));
+  };
+
+  const reset = () => {
+    setMovie(initialMovie);
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    onAdd(movie);
+
+    setCount((currentCount) => currentCount + 1);
+
+    reset();
```

Comment:
you function have a lot of spaces, it can be less
```suggestion
    event.preventDefault();

    onAdd(movie);
    setCount((currentCount) => currentCount + 1);
    reset();
```